src/sentry/replays/usecases/pack.py

```diff
@@ -0,0 +1,53 @@
+"""Packed binary encoding module.
+
+Bytes are packed in varying formats dependent on their type. The type is
+determined by the first byte and is an 8-bit integer. 0 for rrweb, 1 for
+video. For backwards compatibility reasons the maximum type byte allowed
+is 90 (91 being the ascii encoding for `[` which is the leading character
+of a JSON array).
+
+The RRWeb type exclusively contains rrweb json after the type byte. The
+video type contains an 8-byte length header followed by video bytes
+followed by rrweb bytes. The length bytes encode offset information
+detailing how to split the rrweb payload from the video payload.
+
+A type byte is not always specified. Past encodings will lead with the
+binary encoding for the `[` character. These are considered rrweb type
+payloads and can be returned as is.
+"""
+
+from enum import Enum
+
+
+class Encoding(Enum):
+    RRWEB = 0
+    VIDEO = 1
+
+
+def pack(rrweb: bytes, video: bytes | None) -> bytes:
+    def _to_u64_bytes(length):
```

Comment:
max value of a 32 bit would be 2.4 GB video, right? In that case, would that be enough instead of 64?

Reply:
Yeah we could.  Could probably get away with a 24-bit integer too.

Reply:
if we're going unsigned integer (makes sense) and it's the max size of a single segment, indeed could be quite small. Anyway no relevant gains, it's just a single int 😅